presto-main/src/main/java/com/facebook/presto/server/SessionContext.java

```diff
@@ -56,6 +57,11 @@
     @Nullable
     String getLanguage();
 
+    @Nullable
+    Optional<Tracer> getTracer();
+
+    void setTracerIfEnabled(Tracer tracer);
```

Comment:
this needs some thinking. Most classes in Presto are immutable. We need avoid using setters. Can we set the tracer in constructor?

Reply:
Yeah we can set the tracer in the constructor. We'll need to add a new constructor to HttpRequestSessionContext that takes in an additional parameter tracer. And inside we still have to manually match the session property enable_distributed_tracing to decide if we want to setup the passed in tracer or noop tracer.